UnityProject/Assets/Scripts/Objects/Directionals/DirectionalPassable.cs

```diff
@@ -1,105 +1,210 @@
-		if (!Passable) return false;
-
-		Vector2Int sideToCross = (sideVector - LocalPosition).To2Int();
-		bool result = IsPassableAtSide(sideToCross);
-		return result;
-	}
+			return IsPassableAtSide(GetSideFromVector(leavingTo), leavableSides);
+		}
 
-	bool IsPassableAtSide(Vector2Int sideToCross)
-	{
-		// TODO Consider using some data structure
-		switch (Directional.CurrentDirection.AsEnum())
+		public override bool IsAtmosPassable(Vector3Int enteringFrom, bool isServer)
 		{
-			case OrientationEnum.Up:
-				if (sideToCross == Vector2Int.up) return PassableDown;
-				else if (sideToCross == Vector2Int.down) return PassableUp;
-				else if (sideToCross == Vector2Int.left) return PassableRight;
-				else if (sideToCross == Vector2Int.right) return PassableLeft;
-				break;
-			case OrientationEnum.Down:
-				if (sideToCross == Vector2Int.up) return PassableUp;
-				else if (sideToCross == Vector2Int.down) return PassableDown;
-				else if (sideToCross == Vector2Int.left) return PassableLeft;
-				else if (sideToCross == Vector2Int.right) return PassableRight;
-				break;
-			case OrientationEnum.Left:
-				if (sideToCross == Vector2Int.up) return PassableLeft;
-				else if (sideToCross == Vector2Int.down) return PassableRight;
-				else if (sideToCross == Vector2Int.left) return PassableDown;
-				else if (sideToCross == Vector2Int.right) return PassableUp;
-				break;
-			case OrientationEnum.Right:
-				if (sideToCross == Vector2Int.up) return PassableRight;
-				else if (sideToCross == Vector2Int.down) return PassableLeft;
-				else if (sideToCross == Vector2Int.left) return PassableUp;
-				else if (sideToCross == Vector2Int.right) return PassableDown;
-				break;
+			if (!AtmosPassable) return false;
+			if (IsAtmosPassableOnAll) return true;
+
+			return IsPassableAtSide(GetSideFromVector(enteringFrom), atmosphericPassableSides);
+		}
+
+		#endregion RegisterObject Overrides
+
+		/// <summary>
+		/// Sets all sides for this object as passable for the given pass type.
+		/// </summary>
+		public void AllowPassableOnAllSides(PassType passType)
+		{
+			switch (passType)
+			{
+				case PassType.Entering:
+					isEnterableOnAll = true;
+					Passable = true;
+					break;
+				case PassType.Leaving:
+					isLeavableOnAll = true;
+					Passable = true;
+					break;
+				case PassType.Atmospheric:
+					isAtmosPassableOnAll = true;
+					AtmosPassable = true;
+					break;
+				default:
+					break;
+			}
+		}
+
+		/// <summary>
+		/// Sets all sides for this object as impassable for the given pass type.
+		/// </summary>
+		public void DenyPassableOnAllSides(PassType passType)
+		{
+			switch (passType)
+			{
+				case PassType.Entering:
+					isEnterableOnAll = false;
+					Passable = false;
+					break;
+				case PassType.Leaving:
+					isLeavableOnAll = false;
+					Passable = false;
+					break;
+				case PassType.Atmospheric:
+					isAtmosPassableOnAll = false;
+					AtmosPassable = false;
+					break;
+				default:
```

Comment:
Codacy found an issue: [Remove this empty 'default' clause.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6377637)